File: Paid_ChatGPT/sistemaDeTipos/tipos.js
function calcularPromedioMayoresAUmbral(precios, umbral) {
    let suma = 0;
    let cantidad = 0;

    for (const precio of precios) {
        if (precio > umbral) {
            suma += precio;
            cantidad++;
        }
    }

    return cantidad === 0 ? 0 : suma / cantidad;
}

// Ejemplo de uso
const precios = ["10", "20", "30", "40", "50"];
const promedio = calcularPromedioMayoresAUmbral(precios, 25);

console.log(promedio); // 40

File: Paid_ChatGPT/sistemaDeTipos/tipos.rs
fn calcular_promedio_mayores_a_umbral(precios: &[f64], umbral: f64) -> f64 {
    let mut suma = 0.0;
    let mut cantidad = 0usize;

    for &precio in precios {
        if precio > umbral {
            suma += precio;
            cantidad += 1;
        }
    }

    if cantidad == 0 {
        0.0
    } else {
        suma / cantidad as f64
    }
}

fn main() {
    let precios = [10.0, 20.0, 30.0, 40.0, 50.0];

    let promedio = calcular_promedio_mayores_a_umbral(&precios, 25.0);

    println!("{:.2}", promedio); // 40.00
}